problem description: when warning reporting is disabled, the compilation was
failing with undefined symbols

Index: git/Source/core/CallsignTLS.cpp
===================================================================
--- git.orig/Source/core/CallsignTLS.cpp
+++ git/Source/core/CallsignTLS.cpp
@@ -20,7 +20,7 @@
 #include "CallsignTLS.h"
 #include "Thread.h"
 
-#ifdef __CORE_WARNING_REPORTING__
+#if defined(__CORE_WARNING_REPORTING__) || defined(__CORE_EXCEPTION_CATCHING__)
 
 namespace WPEFramework {
 namespace Core {
